test/dynamo/test_python_dispatcher.py

```diff
@@ -0,0 +1,77 @@
+# Owner(s): ["module: dynamo"]
+import unittest
+
+import torch
+import torch._dynamo.test_case
+from torch._dispatch.python import enable_python_dispatcher
+from torch._dynamo.testing import CompileCounter, EagerAndRecordGraphs, normalize_gm
+from torch.testing._internal.common_cuda import TEST_CUDA
+
+
+class PythonDispatcherTests(torch._dynamo.test_case.TestCase):
+    def test_dispatch_key(self):
+        eager = EagerAndRecordGraphs()
+
+        @torch.compile(backend=eager, fullgraph=True)
+        def fn(x):
+            key_set = torch._C._dispatch_tls_local_include_set()
+            key_set = key_set | torch._C._dispatch_keys(x)
+            key_set = key_set - torch._C._dispatch_tls_local_exclude_set()
+            if key_set.highestPriorityTypeId() == torch.DispatchKey.PythonDispatcher:
+                return torch.sin(x + 1)
+            else:
+                return torch.sin(x - 1)
+
+        x = torch.randn(2, 3)
+        with enable_python_dispatcher():
+            self.assertEqual(fn(x), torch.sin(x + 1))
```

Comment:
Can you confirm that without `enable_python_dispatcher`, the result is `torch.sin(x - 1)`?

Reply:
Updated!